source/features/clean-conversation-headers.tsx

```diff
@@ -26,25 +30,31 @@ function initIssue(): void {
 
 function initPR(): void {
 	const observer = observe('.gh-header-meta .flex-auto:not(.rgh-clean-conversation-header)', {
-		add(byline) {
+		async add(byline) {
 			byline.classList.add('rgh-clean-conversation-header');
-			const isSameAuthor = select('.js-discussion > .TimelineItem:first-child .author')?.textContent === select('.author', byline)!.textContent;
-			const baseBranch = select('.commit-ref:not(.head-ref)', byline)!;
-			const isDefaultBranch = (baseBranch.firstElementChild as HTMLAnchorElement).pathname.split('/').length === 3;
 
-			// Removes: [octocat wants to merge 1] commit into github:master from octocat:feature
-			// Removes: [octocat] merged 1 commit into master from feature
-			for (const node of [...byline.childNodes].slice(isSameAuthor ? 0 : 2, pageDetect.isMergedPR() ? 2 : 4)) {
+			const author = select('.author', byline)!;
+			const isSameAuthor = pageDetect.isPRConversation() && author.textContent === select('.TimelineItem .author')!.textContent;
+
+			const base = select('.commit-ref', byline)!;
+			const baseBranch = base.title.split(':')[1];
+			const isDefaultBranch = baseBranch === await getDefaultBranch() || (pageDetect.isClosedPR() && baseBranch === 'master');
```

Comment:
Why replace a synchronous default branch check with an async one?

To improve readability, the previous check could be preserved + we can move the second part to a new constant `wasDefaultBranch`

Reply:
This function should have been cached by other features so I see no issue here. Does this actually make it any shower or something?

Reply:
Alright, let's still split the line in 2 as suggested though